### PR TITLE
tap voice entry for the name creation feature

### PR DESCRIPTION
**Pull request by mohammedhemaid**
The Video of the feature:
https://photos.app.goo.gl/mu5uZKQW3fgRF9KL7

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -61,4 +62,12 @@
 
     <string name="rename_list">Rename List</string>
     <string name="delete_list">Delete List</string>
+
+    <string name="speech_message">Tell us what kind of place you\'re looking for.</string>
```
Comment (Gear61) [inline app/src/main/res/values/strings.xml]: Let's customize the copy-pasted code to match our use case. :)
```suggestion
    <string name="list_name_speech_prompt_message">Tell us what you want to name your list</string>
```

File: app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -60,6 +60,17 @@
...
+
+            <com.joanzapata.iconify.widget.IconTextView
+                android:id="@+id/voice_entry_icon"
+                android:text="@string/voice_icon"
+                android:layout_gravity="end|center_vertical"
+                style="@style/list_item_button"
+                android:layout_marginEnd="60dp"
+                android:layout_marginRight="60dp"
+                android:textSize="24sp"
+                android:foreground="?attr/selectableItemBackgroundBorderless"
```
Comment (Gear61) [inline app/src/main/res/layout/activity_main.xml]: Why selectableItemBackgroundBorderless instead of selectableItemBackground?

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -60,6 +60,17 @@
                 android:layout_height="44dp"
                 android:hint="@string/add_list_hint"
                 />
+
+            <com.joanzapata.iconify.widget.IconTextView
+                android:id="@+id/voice_entry_icon"
+                android:text="@string/voice_icon"
+                android:layout_gravity="end|center_vertical"
+                style="@style/list_item_button"
```
Comment (Gear61) [inline app/src/main/res/layout/activity_main.xml]: I don't think we should use this style. We're not within a list and we don't need the centerVertical attribute within it (that's a RelativeLayout only attribute). Let's just inline those attributes.

File: app/src/main/res/layout/activity_main.xml [hunk 4]
```diff
@@ -60,6 +60,17 @@
...
+                android:layout_marginEnd="60dp"
```
Comment (Gear61) [inline app/src/main/res/layout/activity_main.xml]: This should be 55dp I think so there isn't a gap between the icon space and the add button.

File: app/src/main/res/layout/activity_main.xml [hunk 5]
```diff
@@ -60,6 +60,17 @@
                 android:layout_height="44dp"
                 android:hint="@string/add_list_hint"
                 />
+
+            <com.joanzapata.iconify.widget.IconTextView
+                android:id="@+id/voice_entry_icon"
+                android:text="@string/voice_icon"
```
Comment (Gear61) [inline app/src/main/res/layout/activity_main.xml:71]: Let's make the textColor blue so it's more obvious that it's a button.

File: app/src/main/java/com/randomappsinc/studentpicker/home/MainActivity.java [hunk 6]
```diff
@@ -229,19 +242,54 @@ public void onRequestPermissionsResult(
...
-            if (!filePath.endsWith(".txt")) {
-                UIUtils.showSnackbar(parent, getString(R.string.invalid_file));
-            } else {
-                Intent intent = new Intent(this, ImportFileActivity.class);
-                intent.putExtra(ImportFileActivity.FILE_PATH_KEY, filePath);
-                startActivity(intent);
-            }
+
+        switch (requestCode) {
+
```
Comment (Gear61) [inline app/src/main/java/com/randomappsinc/studentpicker/home/MainActivity.java]: nit: Delete this newline.

File: app/src/main/java/com/randomappsinc/studentpicker/home/MainActivity.java [hunk 7]
```diff
@@ -229,19 +242,54 @@ public void onRequestPermissionsResult(
...
+                        UIUtils.showSnackbar(parent, getString(R.string.invalid_file));
+                    } else {
+                        Intent intent = new Intent(this, ImportFileActivity.class);
+                        intent.putExtra(ImportFileActivity.FILE_PATH_KEY, filePath);
+                        startActivity(intent);
+                    }
+                }
+                break;
         }
+
```
Comment (Gear61) [inline app/src/main/java/com/randomappsinc/studentpicker/home/MainActivity.java]: nit: Delete this newline.

**Review by Gear61 — CHANGES_REQUESTED** [review]
I think we'll need to increase the marginRight on the EditText as well (I can't comment on lines not in the PR for some reason) so if the user types a lot, the text doesn't overlap with the new speech input button (as I mentioned in class, we need to always think about the edge cases!). I think we'll need to increase it to 99dp (55dp + 44dp). Just make sure to test the case to make sure it works.

File: app/src/main/res/layout/activity_main.xml (hunk 2 above)
Comment (mohammedhemaid) [reply app/src/main/res/layout/activity_main.xml]: yes, there is no need for selectableItemBackgroundBorderless, I just wanted the animation to be circular but it didn't work

File: app/src/main/res/layout/activity_main.xml (hunk 3 above)
Comment (mohammedhemaid) [reply app/src/main/res/layout/activity_main.xml]: image

image
this is the difference between the center and without the center or centerVertical

Comment (Gear61) [reply app/src/main/res/layout/activity_main.xml]: Yeah, we still need the layout_gravity of center_vertical|end (for FrameLayouts), but we don't need the one within the style (for RelativeLayouts).

Comment (mohammedhemaid) [reply app/src/main/res/layout/activity_main.xml]: Got it, I saw it but I think it will not make a difference so I kept it, but putting them inline it will be better and it will reduce the unused code.

**Comment by mohammedhemaid** [comment]
I also cannot comment on the editText!
I changed it like this: 

image

The video of the clicking:
https://photos.app.goo.gl/MY3eEKxcutoe3akz5

File: app/src/main/res/layout/activity_main.xml [hunk 8]
```diff
@@ -53,13 +53,27 @@
                 android:textColor="@color/dark_gray"
                 android:textSize="18dp"
                 android:background="@drawable/add_item_border"
-                android:layout_marginRight="55dp"
+                android:layout_marginRight="97dp"
```
Comment (Gear61) [inline app/src/main/res/layout/activity_main.xml]: Thank you so much for the video! That is not the look I was going for; I suggested the wrong code, sorry! :(

We should keep the 55dp marginRight so the input doesn't overlap with the button but add a paddingRight of 44dp as well so the typed in text doesn't overlap with the microphone icon (the icon is still within the text input).

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -61,4 +62,12 @@
 
     <string name="rename_list">Rename List</string>
     <string name="delete_list">Delete List</string>
+
+    <string name="speech_message">Tell us what you want to name your list</string>
```
Comment (Gear61) [inline app/src/main/res/values/strings.xml]: speech_message is too generic, and we want to add speech input to other parts of the app.
```suggestion
    <string name="list_name_input_speech_message">Tell us what you want to name your list</string>
```

**Review by Gear61 — CHANGES_REQUESTED** [review]
Thanks for addressing the feedback! Just a couple more changes, and we should be good to go.

File: app/src/main/res/layout/activity_main.xml (hunk 8 above)
Comment (mohammedhemaid) [reply app/src/main/res/layout/activity_main.xml]: The video of the update:
https://photos.app.goo.gl/wVeuWHSdMDKFrtNP8

**Review by Gear61 — APPROVED** [review]
Looks good! Merging this in now. :)
